### ISSUE TITLE
eclipse.mavencentral with underspecified releases

### ISSUE THREAD

**Issue by acisternino** [diffplug/goomph#121]
`com.diffplug.eclipse.mavencentral` fails immediately when gradle is invoked.

The error happens when the plugin tries to download the `https://download.eclipse.org/eclipse/updates/xx.xx/artifacts.jar` archive from eclipse.org:

```
11:01 $ ./gradlew clean

FAILURE: Build failed with an exception.

* Where:
Build file '/home/andrea/Development/skunkworks/java-swt-gradle/build.gradle' line: 23

* What went wrong:
A problem occurred evaluating root project 'java-swt-gradle'.
> https://download.eclipse.org/eclipse/updates/4.14/artifacts.jar
  received http code 404
  <!DOCTYPE html>
  ... Rest of the HTML follows ...
```

## Steps to reproduce

1. Clone minimal sample project: https://github.com/acisternino/java-swt-gradle
1. Execute any Gradle task using the wrapper. E.g. `./gradlew clean`

## Environment

- Java 11
- Gradle 6.3 (using the wrapper)
- Several Eclipse versions (tested with 4.7, 4.11, 4.14, 4.15)
- Plugin version 3.22.0

**Comment by nedtwigg** [comment]
Any chance you're behind a proxy?

**Comment by nedtwigg** [comment]
Just FYI, I tried this example project on my machine, and it worked fine (I don't have Java 11, so the compile failed, but the SWT jars downloaded fine).  I bet this is a firewall/proxy issue.  This is where we download the "artifacts.jar" from the official eclipse p2 repository, which is what we use to determine version numbers for SWT, JFace, etc.

https://github.com/diffplug/goomph/blob/a5849ddc8475524fdec593d326959e458a4c47c5/src/main/java/com/diffplug/gradle/eclipse/MavenCentralMapping.java#L98

And this is how the download happens:

https://github.com/diffplug/goomph/blob/a5849ddc8475524fdec593d326959e458a4c47c5/src/main/java/com/diffplug/gradle/FileMisc.java#L54-L72

Happy to [take a PR](https://github.com/diffplug/goomph/blob/master/CONTRIBUTING.md#test-locally) which fixes your issue, but it's very hard to debug proxy issues from outside the proxy.

**Comment by acisternino** [comment]
Unfortunately I am not behind any proxy.
I have just tested again from my home PC directly connected to the internet through a regular ADSL modem/router.

What is perplexing is that the URL printed in the error report (`https://download.eclipse.org/eclipse/updates/4.14/artifacts.jar`) doesn't exist. If I try it in the browser I get a **Not Found** page that matches Gradle's output.

Going to `https://download.eclipse.org/eclipse/updates/4.14/` show a page saying that I have hit an **Eclipse software repository** whose content is:

- R-4.14-201912100610/
- categories/
- compositeArtifacts.jar
- compositeContent.jar
- p2.index

[error.html.gz](https://github.com/diffplug/goomph/files/4417673/error.html.gz)

**Comment by acisternino** [comment]
I have just noticed that the URL of the `artifacts.jar` file actually is:

`http://download.eclipse.org/eclipse/updates/4.14/R-4.14-201912100610/artifacts.jar`

i.e. one level below 4.14.

This happens also in more recent versions. The actual file is in the directory with the release number and the timestamp.

**Comment by nedtwigg** [comment]
Aha!  In that case, in your build, try changing `4.14` to `4.14.0`.  Let me know if that fixes it, in which case this will become a bug that we fix in the next version.

**Comment by acisternino** [comment]
Thanks @nedtwigg !

It worked after adding the third number to the version. Sorry for not testing this before.

Maybe a note in the documentation could be helpful to avoid the same mistake in the future.

**Comment by nedtwigg** [comment]
> Maybe a note in the documentation

I think it's a bug that we allow you to try at all.  Next version will give a warning "You have to be specific, instead of `4.14` use `4.14.0`"

**Comment by nedtwigg** [comment]
Fixed in `3.23.0`
